pymc3/plots.py

```diff
@@ -168,34 +167,53 @@ def autocorrplot(trace, vars=None, max_lag=100, burn=0, ax=None):
         
     Returns
     -------
-
     ax : matplotlib axes
-
     """
     
     import matplotlib.pyplot as plt
-
+        
+    def _handle_array_varnames(val):
+        #val = str(val)
+        if trace[0][val].__class__ is np.ndarray:
+            k = trace[val].shape[1]
+            for i in xrange(k):
+                yield val + '_{0}'.format(i)
+        else:
+            yield val
+            
     if vars is None:
-        vars = trace.varnames
+        vars = [item  for sub in [[i for i in _handle_array_varnames(var)]
+                    for var in trace.varnames] for item in sub]
     else:
-        vars = [str(var) for var in vars]
```

Comment:
I think its partly the same thing, you need to make sure you're dealing with strings when you think you're dealing with strings.